### PR TITLE
Add support for java 11 functions

### PR DESCRIPTION
**Pull request by Flanker32**
Add support for java 11 functions

### CODE REVIEW

**Comment by andxu** [comment]
Add telemetry for java 8 or 11.

File: azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/WindowsFunctionRuntimeHandler.java [hunk 1]
```diff
@@ -59,11 +66,25 @@ protected WindowsFunctionRuntimeHandler(Builder builder) {
...
         }
+        appWithCreate.withJavaVersion(javaVersion).withWebContainer(null);
         return appWithCreate;
     }
 
     @Override
     public FunctionApp.Update updateAppRuntime(FunctionApp app) {
-        return app.update();
+        final JavaVersion serverJavaVersion = app.javaVersion();
+        final FunctionApp.Update update = app.update();
```
Comment (andxu) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/WindowsFunctionRuntimeHandler.java:76]: handle null serverJavaVersion

File: azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/FunctionRuntimeHandler.java [hunk 2]
```diff
@@ -72,4 +80,26 @@ protected void changeAppServicePlan(FunctionApp app, AppServicePlan appServicePl
     protected ResourceGroup getResourceGroup() {
         return azure.resourceGroups().getByName(resourceGroup);
     }
+
+    protected JavaVersion parseJavaVersion() {
```
Comment (andxu) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/FunctionRuntimeHandler.java:84]: protected to private

File: azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/WindowsFunctionRuntimeHandler.java [hunk 3]
```diff
@@ -59,11 +66,25 @@ protected WindowsFunctionRuntimeHandler(Builder builder) {
...
             }
         }
+        appWithCreate.withJavaVersion(javaVersion).withWebContainer(null);
         return appWithCreate;
     }
 
     @Override
     public FunctionApp.Update updateAppRuntime(FunctionApp app) {
-        return app.update();
+        final JavaVersion serverJavaVersion = app.javaVersion();
```
Comment (andxu) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/WindowsFunctionRuntimeHandler.java:75]: when local configuration has not configured java version and shall we update Java version at server side?  

```
if (javaVersion.equals(serverJavaVersion)) {
            Log.info(String.format(HOST_JAVA_VERSION, serverJavaVersion));
        } else if (JavaVersion.OFF.equals(serverJavaVersion)) {
            Log.info(HOST_JAVA_VERSION_OFF);
            update.withJavaVersion(javaVersion);
        }
```

Comment (Flanker32) [reply azure-tools-common/src/main/java/com/microsoft/azure/common/function/handlers/runtime/WindowsFunctionRuntimeHandler.java:75]: Will update the server side java version only if user set javaVersion in `pom.xml`
``` java
else if (StringUtils.isNotEmpty(runtimeConfiguration.getJavaVersion())) {
            // Will update function host java version if user specify it in pom
            Log.info(String.format(HOST_JAVA_VERSION_UPDATE, serverJavaVersion, javaVersion));
            update.withJavaVersion(javaVersion);
        }
```

**Pull request merged**